storage/s3_storage.py

```diff
@@ -73,13 +74,24 @@ def save_to_directory(self, directory_path: str) -> None:
     def load_from_filename(self, file_path: str) -> None:
         client = self._connect()
 
+        extra_args = None
+        content_type = mimetypes.guess_type(file_path)[0]
+        if content_type is not None:
+            extra_args = {"ContentType": content_type}
```

Comment:
Since this code is being used in a few places, should we refactor this out into its own function?

Reply:
Not sure what the interface to that function would be? 

Note that the code is slightly different for each iteration, because the AWS functions all take slightly different arguments (one takes `ContentType` directly as a keyword argument, another takes `ExtraArgs` as a dictionary and another takes `extra_args` as a dictionary. Some expect `None` when there's nothing to set, and some expect an empty dictionary.

Reply:
I was thinking something like this:
```
def get_content_type(file_path, default_value):
        extra_args = default_value
        content_type = mimetypes.guess_type(file_path)[0]
        if content_type is not None:
            extra_args = {"ContentType": content_type}
        return extra_args
```
And it could be called like:
`extra_args = get_content_type(file_path, None)`
`extra_args = get_content_type(file_path, {})`

Unless I'm missing something, that should work for all scenarios.

Reply:
To be honest, I, personally, find that harder to read. While there are fewer lines of code, the function is overloaded to do different things in different contexts, which makes it harder to understand what is happening/should happen. It also seems like it'd be easier to have it do the "wrong thing", if we change it in the future. (also, sometimes you'd need to pass the local file path, and sometimes the storage URI, which is another slight difference between the implementations).

Personally, I'd rather invest effort into using the same AWS methods for the different upload functions (at least the functions that upload from a local file path, if not from a file-like-object).

Reply:
I didn't realize that we might need to also pass in the storage URI.
Yep I agree that using the same AWS methods would be be better where we can.